semtypes/src/main/java/io/ballerina/types/subtypedata/TableSubtype.java

```diff
@@ -29,8 +38,17 @@
  */
 public class TableSubtype {
 
-    public static SemType tableContaining(SemType memberType) {
-        Bdd bdd = (Bdd) Core.subtypeData(memberType, BasicTypeCode.BT_MAPPING);
-        return Core.createBasicSemType(BasicTypeCode.BT_TABLE, bdd);
+    public static SemType tableContaining(Env env, SemType mappingType, CellAtomicType.CellMutability mut) {
```

Comment:
Would it be better to make this class final (so people can't extend and overload any of these methods) and give a private constructor (to prevent people from creating instances of `TableSubtype`)? granted this is not practice we have fallowed consistently with other `*Subtype` classes (some of them have private constructors and some of them are final)